dev_guide/deployments.adoc

```diff
@@ -396,14 +419,34 @@ no strategy is specified on a deployment configuration.
 [[rolling-strategy]]
 === Rolling Strategy
 
-The rolling strategy performs a rolling update and supports
-xref:lifecycle-hooks[lifecycle hooks] for injecting code into the deployment
-process.
+A rolling deployment slowly replaces instances of the previous version of an application
+with instances of the new version of the application. A rolling deployment typically waits
+for new pods to become *ready* via a *readiness check* before scaling down the old components.
+If a significant issue occurs, the rolling deployment can be aborted.
 
-The rolling deployment strategy waits for pods to pass their
-xref:../dev_guide/application_health.adoc#dev-guide-application-health[readiness
-check] before scaling down old components, and does not allow pods that do not
-pass their readiness check within a configurable timeout.
+=== Canary checks
+
+All rolling deployments in {product-title} are *canary* deployments - a new version (the canary)
+is tested  before all of the old instances are replaced. If the readiness check never succeeds,
+the canary instance is removed and the deployment configuration will be automatically rolled back.
+The readiness check is part of the application code, and may be as sophisticated as necessary to
+ensure the new instance is ready to be used. If you need to implement more complex checks of the
+application (such as sending real user workloads to the new instance), consider implementing a
+custom deployment or using a blue-green deployment strategy.
+
+
+=== When to use a rolling deployment?
```

Comment:
should we also have `When to not use a rolling deployment?` (like when you have RWO volume?)